Tests/TestUtils/MockNetworkService.swift

```diff
@@ -58,7 +56,7 @@ class MockNetworkService: Networking {
     /// Sets the mock `HttpConnection` response connection for a given `NetworkRequest`. Should only be used
     /// when in mock mode.
     func setMockResponseFor(networkRequest: NetworkRequest, responseConnection: HttpConnection?) {
-        helper.setResponseFor(networkRequest: networkRequest, responseConnection: responseConnection)
+        helper.addResponseFor(networkRequest: networkRequest, responseConnection: responseConnection ?? defaultMockResponse(networkRequest.url))
```

Comment:
Adding a default isn't strictly needed as because of the `else` clause in the `if let response ...` statement above on line 37.